### PR TITLE
[metadata] Fix resource leak, on error, we did not free the named arguments.  Coverity 1443346

### PR DESCRIPTION
**Pull request by migueldeicaza**
[metadata] Fix resource leak, on error, we did not free the named arguments. Coverity 1443346

### CODE REVIEW

File: mono/metadata/custom-attrs.c [hunk 1]
```diff
@@ -1316,6 +1316,7 @@ mono_reflection_create_custom_attr_data_args_noalloc (MonoImage *image, MonoMeth
 fail:
 	mono_error_set_generic_error (error, "System.Reflection", "CustomAttributeFormatException", "Binary format of the specified custom attribute was invalid.");
 	g_free (arginfo);
+	g_free (named_args);
```
Comment (akoeplinger) [inline mono/metadata/custom-attrs.c]: I don't see where we're allocating `named_args` in this function? isn't it the responsibility of the caller?

Comment (jaykrell) [reply mono/metadata/custom-attrs.c]: Presumably it is supposed to be `namedargs` without the underscore.
https://github.com/mono/mono/pull/13855/files#diff-7781be09cd2ddfae798d13ecb23a63c6R1224

I suggest, um, renaming the args. `out_namedargs` and `namedargs`?

Comment (jaykrell) [reply mono/metadata/custom-attrs.c]: And `namedargs` will need null initialization also.

Comment (jaykrell) [reply mono/metadata/custom-attrs.c]: Or put a star on it and ignore the rest I said, `g_free(*named_args);` that should do, but double check.

Comment (jaykrell) [reply mono/metadata/custom-attrs.c]: No the last point is bad. We don't write the out parameter until the very end of success. So the first part -- null initialize the local and add underscore to the free.

**Comment by CoffeeFlux** [comment]
I think it's worth doing some variable naming throughout the file to avoid this kind of ambiguity and to ensure everything is properly NULL-initialized and freed. I'll push to this branch with that work and a fix for the specific issue coverity caught.

**Comment by CoffeeFlux** [comment]
Wasn't totally sure on naming, but other stuff in the file seemed to use underscores so I just switched all the *args to that. Should be easier to follow the logic now.

**Comment by CoffeeFlux** [comment]
@monojenkins build failed

**Pull request merged**